ui/src/shared/components/TableGraph.js

```diff
@@ -118,34 +121,62 @@ class TableGraph extends Component {
 
     this.setState({
       data,
+      labels,
       processedData,
+      sortedTimeVals,
       sortField: sortFieldName,
       sortDirection: direction,
     })
   }
 
-  calcHoverTimeIndex = (data, hoverTime, verticalTimeAxis) => {
-    if (_.isEmpty(data) || hoverTime === NULL_HOVER_TIME) {
-      return undefined
+  calcScrollToColRow = () => {
+    const {
+      data,
+      sortedTimeVals,
+      hoveredColumnIndex,
+      hoveredRowIndex,
+    } = this.state
+    const {hoverTime, tableOptions} = this.props
+    if (_.isEmpty(data[0]) || hoverTime === NULL_HOVER_TIME) {
+      return {scrollToColumn: undefined, scrollToRow: undefined}
     }
-    if (verticalTimeAxis) {
-      return data.findIndex(
-        row => row[0] && _.isNumber(row[0]) && row[0] >= hoverTime
-      )
+    const hoveringThisTable = hoveredColumnIndex !== NULL_ARRAY_INDEX
+    if (hoveringThisTable) {
+      return {scrollToColumn: hoveredColumnIndex, scrollToRow: hoveredRowIndex}
     }
-    return data[0].findIndex(d => _.isNumber(d) && d >= hoverTime)
+
+    const {verticalTimeAxis} = tableOptions
+
+    const hoverTimeIndex = sortedTimeVals.reduce(
+      (acc, currentTime, index, array) => {
+        const diff = Math.abs(hoverTime - currentTime)
+        if (diff === 0) {
+          return index
+        }
+        if (Math.abs(hoverTime - array[acc]) > diff) {
+          return index
+        }
+        return acc
+      },
+      1
```

Comment:
should there be a comment or something to explain why we start with 1 instead of 0 in the accumulator?